demo/src/main/java/com/eis/geoCalendar/demo/Behaviour/EventMapBehaviour.java

```diff
@@ -0,0 +1,470 @@
+     */
+    @Override
+    public void onMapLongClick(LatLng latLng) {
+        addEventDialog.setEventPosition(latLng);
+        addEventDialog.show(supportFragmentManager, CREATE_EVENT_DIALOG_TAG);
+    }
+
+    /**
+     * Moves the camera to the marker's location and, if set,
+     * expands the bottomSheet to display Marker's event's description
+     * If a goToNavigatorView was set, it will turn visible
+     * Note: returning false for this event will fire the default event for Google Map
+     *
+     * @param marker The marker that caused this callback
+     * @return True if the listener has consumed the event, false otherwise.
+     */
+    @Override
+    public boolean onMarkerClick(Marker marker) {
+        currentFocusMarker = marker;
+        moveMap(marker.getPosition());
+
+        if (goToNavigatorView != null)
+            goToNavigatorView.setVisibility(View.VISIBLE);
+
+        if (bottomSheetBehaviour != null) {
+            bottomSheetBehaviour.setDisplayedText(marker.getTitle());
+            if (bottomSheetBehaviour.isShown())
+                bottomSheetBehaviour.hide();
+            else
+                bottomSheetBehaviour.show();
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * If set, hides the Bottom Sheet
+     * If set, hides the goToNavigatorView
+     * Makes forget the previously set currentFocusMarker
+     *
+     * @param latLng The position on the map where the user clicked
+     */
+    @Override
+    public void onMapClick(LatLng latLng) {
+        if (bottomSheetBehaviour != null)
+            bottomSheetBehaviour.hide();
+
+        if (goToNavigatorView != null)
+            goToNavigatorView.setVisibility(INVISIBLE);
+
+        currentFocusMarker = null;
+    }
+
+    /**
+     * Called by a fragment of the application that returns the parameters to Create a new event and update the map
+     *
+     * @param pos         The position of the event to create
+     * @param description The description of the event to create
+     */
+    @Override
+    public void onEventReturn(LatLng pos, String description) {
+        GPSPosition eventPos = new GPSPosition(pos.latitude, pos.longitude);
+        Event event = new GenericEvent<>(eventPos, description);
+        if (onEventCreatedListeners != null && !onEventCreatedListeners.isEmpty())
+            for (OnEventCreatedListener listener : onEventCreatedListeners)
+                listener.onEventCreated(event);
+
+        Marker created = mMap.addMarker(new MarkerOptions().position(pos).title(description)); //automatically cuts title if too long
+        created.setTag(event);
+        moveMap(pos);
+    }
+
+    /**
+     * Called when user long presses an InfoWindow of Marker
+     *
+     * @param marker The marker whose InfoWindow was pressed
+     */
+    @Override
+    public void onInfoWindowLongClick(Marker marker) {
+        currentFocusMarker = marker;
+        callRemoveEventDialog(marker);

```

Comment:
I don't get why not put the `currentFosusMarker`, maybe something escapes me...
```suggestion
        callRemoveEventDialog(currentFocusMarker);
```